app/models/dor_object_workflow_status.rb

```diff
@@ -0,0 +1,18 @@
+class DorObjectWorkflowStatus
+  attr_reader :pid
+
+  ##
+  # @param [String] pid in format "druid:abc123def4567"
+  def initialize(pid)
+    @pid = pid
+  end
+
+  ##
+  # @return [Boolean]
+  def can_open_version?
+    return false unless Dor::WorkflowService.get_lifecycle('dor', pid, 'accessioned')
+    return false if Dor::WorkflowService.get_active_lifecycle('dor', pid, 'submitted')
+    return false if Dor::WorkflowService.get_active_lifecycle('dor', pid, 'opened')
+    true
+  end
```

Comment:
It could be. This is a straight copy over method so I'm less inclined to change things here. Also to me (and I realize I might be an outlier) that next conditional is harder to read. Realistically I'll be porting over other logic and this will change to be similar to `WorkflowServiceController` private methods.